### PR TITLE
changes for flag add ScanOnDeployment

### PR DESCRIPTION
**Pull request by v-nitja**

### CODE REVIEW

File: 04-Continous-Assurance/Readme.md [hunk 1]
```diff
@@ -251,6 +253,7 @@ Update-AzSKContinuousAssurance -SubscriptionId <SubscriptionId> `
 |FixRuntimeAccount|Use this switch to fix CA runtime account in case of below issues.<ol><li>Runtime account deleted<br>(Permissions required: Subscription owner)</li><li>Runtime account permissions missing<br>(Permissions required: Subscription owner and AD App owner)</li><li>Certificate deleted/expired<br>(Permissions required: Subscription owner and AD App owner)</li></ol>|FALSE|None||
 |FixModules|Use this switch in case 'AzureRm.Automation' module extraction fails in CA Automation Account.|FALSE|None||
 |RenewCertificate|Renews certificate credential of CA SPN if the caller is Owner of the AAD Application (SPN). If the caller is not Owner, a new application is created with a corresponding SPN and a certificate owned by the caller. CA uses the updated credential going forward.|FALSE|None||
+|ScanOnDeployment|CA scan can be auto-triggered upon resource deployment.|FALSE|None||
```
[the remark this answers is not in the document]
Comment (v-nitja) [reply 04-Continous-Assurance/Readme.md]: Will add in the parameter description also , but have already mentioned in section part of the same.

File: 04-Continous-Assurance/Readme.md [hunk 2]
```diff
@@ -708,6 +711,48 @@ The main/dominant component of the cost is automation runtime (storage/OMS costs
      - 0.27+0.07 = $0.34/year
 
 
+[Back to top…](Readme.md#contents)
+
+## Continuous Assurance (CA) - 'ScanOnDeployment' mode
+
```
Comment (ShrutiBhambhani) [inline 04-Continous-Assurance/Readme.md]: Title can be more user friendly - CA - Trigger scan on resource deployment

Comment (sbyna-msft) [reply 04-Continous-Assurance/Readme.md]: Make the title more firendly

File: 04-Continous-Assurance/Readme.md [hunk 3]
```diff
@@ -11,6 +11,7 @@
 - [Removing a Continuous Assurance setup](Readme.md#removing-a-continuous-assurance-setup)
 - [Getting details about a Continuous Assurance setup](Readme.md#getting-details-about-a-continuous-assurance-setup)
 - [Continuous Assurance (CA) - 'Central Scan' mode](Readme.md#continuous-assurance-ca---central-scan-mode)
+- [Continuous Assurance (CA) - 'ScanOnDeployment' mode](Readme.md#continuous-assurance-ca---scanondeployment-mode)
```
[the remark this answers is not in the document]
Comment (v-nitja) [reply 04-Continous-Assurance/Readme.md]: As discussed with Manish, we should mark it as preview and track down the changes required for "GA".

**Pull request merged**